parsec/_parsec_pyi/local_manifest.pyi

```diff
@@ -0,0 +1,399 @@
+from typing import Any, Dict, FrozenSet, Optional, Set, Tuple, Pattern, Union
+
+from ..core.fs.workspacefs.file_operations import Chunks
+from .crypto import SecretKey
+from parsec._parsec_pyi.ids import ChunkID, DeviceID, EntryID
+from parsec._parsec_pyi.manifest import (
+    BlockAccess,
+    EntryName,
+    FileManifest,
+    FolderManifest,
+    UserManifest,
+    WorkspaceEntry,
+    WorkspaceManifest,
+)
+from parsec._parsec_pyi.time import DateTime
+
+class Chunk:
+    def __init__(
+        self,
+        id: ChunkID,
+        start: int,
+        stop: int,
+        raw_offset: int,
+        raw_size: int,
+        access: Optional[BlockAccess],
+    ) -> None: ...
+    def __repr__(self) -> str: ...
+    def __eq__(self, other: Union[Chunk, int]) -> bool: ...
+    def __ne__(self, other: Union[Chunk, int]) -> bool: ...
+    def __lt__(self, other: int) -> bool: ...
+    def __gt__(self, other: int) -> bool: ...
+    def __le__(self, other: int) -> bool: ...
+    def __ge__(self, other: int) -> bool: ...
+    @property
+    def id(self) -> ChunkID: ...
+    @property
+    def start(self) -> int: ...
+    @property
+    def stop(self) -> int: ...
+    @property
+    def raw_offset(self) -> int: ...
+    @property
+    def raw_size(self) -> int: ...
+    @property
+    def access(self) -> Optional[BlockAccess]: ...
+    def evolve(
+        self,
+        id: Optional[ChunkID] = None,
+        start: Optional[int] = None,
+        stop: Optional[int] = None,
+        raw_offset: Optional[int] = None,
+        raw_size: Optional[int] = None,
+        access: Optional[BlockAccess] = None,
+    ) -> Chunk: ...
+    def evolve_as_block(self, data: bytes) -> Chunk: ...
+    def is_block(self) -> bool: ...
+    def is_pseudo_block(self) -> bool: ...
+    def get_block_access(self) -> BlockAccess: ...
+    @classmethod
+    def from_block_access(cls, block_access: BlockAccess) -> Chunk: ...
+    @classmethod
+    def new(cls, start: int, stop: int) -> Chunk: ...
+
+class LocalFileManifest:
```

Comment:
We can cheat here by declaring the manifest inherits BaseManifest:
```python
from parsec.api.data import BaseManifest as BaseRemoteManifest
[...]
class FolderManifest(BaseRemoteManifest):
    [...]
```

This fine given it's only taken into account during typing (so zero risk this will break something at runtime)

Reply:
As you said, it's a cheat. For me it's better to rework the python code to not rely on the inheritance of `BaseRemoteManifest`